File: examples/dev/ir-spec/import_enum_in_enum.ml
type maybe_pair'0 = maybe_pair

and maybe_pair =
  | Maybe_pair'Really_pair of (int, int) Mod03.pair'2
  | Maybe_pair'Empty

let rec mk_pair'2 : int * int -> (int, int) Mod03.pair'2 = function
  | v_A, v_B -> Mod03.Pair'Pair (v_A, v_B)

let rec mk_ordered_pair'1 : (int, int) Mod03.pair'2 -> maybe_pair'0 = function
  | Mod03.Pair'Pair (v_A, v_B) as v_P when v_A >= v_B ->
      Maybe_pair'Really_pair v_P
  | Mod03.Pair'Pair (v_A, v_B) when v_A < v_B -> Maybe_pair'Empty

File: examples/dev/ir-spec/import_enum_in_enum.erl
-file("dev/src/import_enum_in_enum.erlt", 1).

-module(import_enum_in_enum).

-compile([warn_unused_import, warnings_as_errors]).

-export([mk_ordered_pair/1, mk_pair/2]).

-export_type([maybe_pair/0]).

-import_type({mod03, [{pair, 2}]}).

-type maybe_pair() :: {969696,
                       import_enum_in_enum,
                       maybe_pair,
                       really_pair,
                       mod03:pair(integer(), integer())} |
                      {969696, import_enum_in_enum, maybe_pair, empty}.

-spec mk_pair(integer(),
              integer()) -> mod03:pair(integer(), integer()).

mk_pair(A, B) -> {969696, mod03, pair, pair, A, B}.

-spec mk_ordered_pair(mod03:pair(integer(),
                                 integer())) -> maybe_pair().

mk_ordered_pair({969696, mod03, pair, pair, A, B} = P)
    when A >= B ->
    {969696,
     import_enum_in_enum,
     maybe_pair,
     really_pair,
     P};
mk_ordered_pair({969696, mod03, pair, pair, A, B})
    when A < B ->
    {969696, import_enum_in_enum, maybe_pair, empty}.


